src/App.tsx

```diff
@@ -15,61 +14,185 @@ const goodsFromServer = [
   'Garlic',
 ];
 
+const minLengths = new Array(10).fill(0).map((_, i) => i + 1);
+
 enum SortType {
   NONE,
-  ALPABET,
+  ALPHABET,
   LENGTH,
 }
 
-// Use this function in the render method
-function getReorderedGoods(
-  goods: string[],
-  sortType: SortType,
-  isReversed: boolean,
-) {
-  // Not to mutate the original array
-  const visibleGoods = [...goods];
-
-  // Sort and reverse goods if needed
-  // ...
-
-  return visibleGoods;
-}
-
-// DON'T save goods to the state
 type State = {
   isStarted: boolean,
   isReversed: boolean,
   sortType: SortType,
+  selected: number,
+  isSortedByLength: boolean,
+  isSortedAlphabetically: boolean,
 };
 
-export const App = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+export class App extends Component<{}, State> {
+  state = {
+    isStarted: false,
+    isReversed: false,
+    sortType: SortType.NONE,
+    selected: 1,
+    isSortedByLength: false,
+    isSortedAlphabetically: false,
+  };
+
+  getReorderedGoods = (
+    goods: string[],
+    sortType: SortType,
+    isReversed: boolean,
+    selected: number,
+    isSortedByLength: boolean,
+    isSortedAlphabetically: boolean,
+  ) => {
+    const visibleGoods = goods.filter(good => good.length <= selected);
+
+    // f equal first word and s equal second word
+    visibleGoods.sort((f, s) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return isSortedAlphabetically
+            ? f.localeCompare(s)
+            : s.localeCompare(f);
+        case SortType.LENGTH:
+          return isSortedByLength
+            ? f.length - s.length
+            : s.length - f.length;
+        default:
+          return 0;
+      }
+    });
+
+    return isReversed
+      ? visibleGoods.reverse()
+      : visibleGoods;
+  };
+
+  startedSort = () => (
+    this.setState({ isStarted: true })
+  );
+
+  sortByLength = () => (
+    this.setState((state) => ({
+      sortType: SortType.LENGTH,
+      isSortedByLength: !state.isSortedByLength,
+    }))
+  );
+
+  sortAlphabetically = () => (
+    this.setState((state) => ({
+      sortType: SortType.ALPHABET,
+      isSortedAlphabetically: !state.isSortedAlphabetically,
+    }))
+  );
+
+  revers = () => (
+    this.setState(state => ({ isReversed: !state.isReversed }))
+  );
+
+  handleChangeSelect = (event: ChangeEvent<HTMLSelectElement>) => (
+    this.setState({ selected: +event.target.value })
```

Comment:
```suggestion
    this.setState({ selected: Number(event.target.value) })
```

Reply:
I agree if you take into account the empty line, but in another way, I think that better used the unary operator